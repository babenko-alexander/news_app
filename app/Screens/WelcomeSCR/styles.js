import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({
	primaryText: {
		color: '$primaryColor',
		borderColor: '$primaryBorderColor',
    fontSize: '$primaryFontSize',
    textAlign: 'center',
    marginBottom: 35
  },
  centeredBox: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#E8F9D0',
    padding: 20
  }
});

export default styles;