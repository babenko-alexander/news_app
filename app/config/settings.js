export const API_KEY = 'e5448c602e864c4da5c3017011f7024a';
export const API_URL = 'https://newsapi.org/v2/top-headlines?country=ua&apiKey=';
export const API_REQUEST = API_URL + API_KEY;

// export default API_REQUEST;