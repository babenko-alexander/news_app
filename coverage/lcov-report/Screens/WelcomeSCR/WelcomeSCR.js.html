<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Screens/WelcomeSCR/WelcomeSCR.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">Screens/WelcomeSCR</a> WelcomeSCR.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">93.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>28/30</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>16/18</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from "react";
import { View, Text } from "react-native";
&nbsp;
import styles from "./styles";
&nbsp;
import { API_REQUEST } from "../../config/settings";
&nbsp;
import SubmitBTN from "../../Components/Buttons/SubmitBTN";
&nbsp;
export default class WelcomeSCR extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: false
    };
  }
&nbsp;
  componentDidMount() {
    // this.getNews();
    // setInterval(this.getNews, 10000);
    this.props.fetchNewsRequest();
    setInterval(this.props.fetchNewsRequest, 10000);
  }<span class="fstat-no" title="function not covered" ></span>
&nbsp;
  componentDidUpdate() {
    // console.log("STATE=", this.state);
    co<span class="cstat-no" title="statement not covered" >nsole.log("PROPS=", this.props);</span>
  }
&nbsp;
  // getNews = () =&gt; {
  //   fetch(API_REQUEST)
  //     .then(response =&gt; {
  //       return response.json();
  //     })
  //     .then(myJson =&gt; {
  //       console.log('myJson=', myJson);
  //       let authors = [];
&nbsp;
  //       let pureArticles = myJson.articles.map((item, index) =&gt; 
  //       { authors.includes(item.author) ? null : authors.push(item.author);
  //         let articleID =  item.publishedAt + item.url + item.title;
  //         return {
  //         id: articleID, 
  //         author: item.author,
  //         urlToImage: item.urlToImage,
  //         url: item.url,
  //         title: item.title,
  //         description: item.description,
  //         publishedAt: item.publishedAt.split("T")[0] + "   " + item.publishedAt.split("T")[1].split("Z")[0]
  //         }
  //       });
  //       if (authors.length &gt; this.props.authorsArr.length) {
  //         this.props.addAuthorsOK(authors);
  //       }
&nbsp;
  //       // console.log('authors===', authors);
  //       let newsIDs = this.props.newsArr.map(item =&gt; item.id);
  //       let freshNewsIDs = [];
        
  //       if (this.props.newsArr.length === 0) {
  //       this.props.addNewsOK(pureArticles);
  //       return pureArticles;
  //       } else {
  //         let freshNews = pureArticles.filter( item =&gt; { 
  //           if ( !newsIDs.includes(item.id) ) {
  //             freshNewsIDs.push(item.id);
  //             return true
  //           } else {
  //             return false;
  //           }
  //           });
  //         if ( this.props.freshNewsArr.length === 0 ) {
  //           this.props.addFreshNewsOK(freshNews);
  //         } else {
  //          let newFreshNews = this.props.freshNewsArr.filter(item =&gt; !freshNewsIDs.includes(item.id));
  //          this.props.addFreshNewsOK(newFreshNews);
  //         }
  //       }
  //         return;
  //     }) 
  //     .catch(() =&gt; {
  //       this.setState({ error: true });
  //       this.props.addNewsERR();
  //       this.props.addAuthorsERR();
  //       this.props.addFreshNewsERR();
  //     });
  // };
&nbsp;
  render() {
    return (
      &lt;View style={styles.centeredBox}&gt;
        &lt;Text style={styles.primaryText}&gt;
          Welcome to the most honest news application!
        &lt;/Text&gt;
        {SubmitBTN("Get News", () =&gt; this.pr<span class="fstat-no" title="function not covered" >op</span>s.navigatio<span class="cstat-no" title="statement not covered" >n.navigate("NewsStripeSCR"))}</span>
      &lt;/View&gt;
    );
  }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Jul 25 2019 16:09:13 GMT+0300 (Восточная Европа, летнее время)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
