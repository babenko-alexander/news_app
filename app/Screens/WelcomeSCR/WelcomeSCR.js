import React, { Component } from "react";
import { View, Text } from "react-native";

import styles from "./styles";

import { API_REQUEST } from "../../config/settings";

import SubmitBTN from "../../Components/Buttons/SubmitBTN";

export default class WelcomeSCR extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: false
    };
  }

  UNSAFE_componentWillMount() {
    this.getNews();
    setInterval(this.getNews, 10000);
  }

  componentDidUpdate() {
    // console.log("STATE=", this.state);
    console.log("PROPS=", this.props);
  }

  getNews = () => {
    fetch(API_REQUEST)
      .then(response => {
        return response.json();
      })
      .then(myJson => {
        console.log('myJson=', myJson);
        let authors = [];
        let newsIDs = this.props.newsArr.map(item => item.id);
        let freshNewsIDs = [];

        let pureArticles = myJson.articles.map((item, index) => 
        { authors.includes(item.author) ? null : authors.push(item.author);
          let articleID =  item.publishedAt + item.url + item.title;
          return {
          id: articleID, 
          author: item.author,
          urlToImage: item.urlToImage,
          url: item.url,
          title: item.title,
          description: item.description,
          publishedAt: item.publishedAt.split("T")[0] + "   " + item.publishedAt.split("T")[1].split("Z")[0]
          }
        });
        if (authors.length > this.props.authorsArr.length) {
          this.props.addAuthorsOK(authors);
        }

        // console.log('authors===', authors);
        
        if (this.props.newsArr.length === 0) {
        this.props.addNewsOK(pureArticles);
        return pureArticles;
        } else {
          let freshNews = pureArticles.filter( item => { 
            if ( !newsIDs.includes(item.id) ) {
              freshNewsIDs.push(item.id);
              return true
            } else {
              return false;
            }
            });
          if ( this.props.freshNewsArr.length === 0 ) {
            this.props.addFreshNewsOK(freshNews);
          } else {
           let newFreshNews = this.props.freshNewsArr.filter(item => !freshNewsIDs.includes(item.id));
           this.props.addFreshNewsOK(newFreshNews);
          }
        }
          return;
      }) 
      .catch(() => {
        this.setState({ error: true });
        this.props.addNewsERR();
        this.props.addAuthorsERR();
        this.props.addFreshNewsERR();
      });
  };

  render() {
    return (
      <View style={styles.centeredBox}>
        <Text style={styles.primaryText}>
          Welcome to the most honest news application!
        </Text>
        {SubmitBTN("Get News", () => {
          let authors = [];
          this.props.newsArr.map( item => {
            authors.includes(item.author) ? null : authors.push(item.author);
            return item.author;
          });
          this.props.addAuthorsOK(authors);
          this.props.navigation.navigate("NewsStripeSCR");
          return;
        }
        )}
      </View>
    );
  }
}
