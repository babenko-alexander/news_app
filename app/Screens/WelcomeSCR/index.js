import { connect } from 'react-redux';
import WelcomeSCR from './WelcomeSCR.js';

// import { addNewsOK, addNewsERR } from '../../store/reducers/newsArr';
// import { addAuthorsOK, addAuthorsERR } from '../../store/reducers/authorsArr';
// import { addFreshNewsOK, addFreshNewsERR } from '../../store/reducers/freshNewsArr';

import { fetchNewsRequest } from '../../store/sagas/actions';

function MSTP(state) {
  return {
    newsArr: state.newsArr,
    freshNewsArr: state.freshNewsArr,
    authorsArr: state.authorsArr
  };
}

function MDTP(dispatch) {
  return {
    fetchNewsRequest: () => dispatch(fetchNewsRequest())
    // addNewsOK: data => dispatch(addNewsOK(data)),
    // addNewsERR: () => dispatch(addNewsERR()),
    // addAuthorsOK: data => dispatch(addAuthorsOK(data)),
    // addAuthorsERR: () => dispatch(addAuthorsERR()),
    // addFreshNewsOK: data => dispatch(addFreshNewsOK(data)),
    // addFreshNewsERR: () => dispatch(addFreshNewsERR()),
  };
}

export default connect(
  MSTP,
  MDTP
)(WelcomeSCR);