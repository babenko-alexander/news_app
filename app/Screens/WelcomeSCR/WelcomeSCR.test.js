import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import WelcomeSCR from './WelcomeSCR';
import 'isomorphic-fetch';

import { API_REQUEST } from '../../config/settings';


describe('WelcomeSCR component', () => {
  it('renders correctly', () => {
    renderer.create(<WelcomeSCR />);
  });
  it('should match to snapshot', () => {
    const component = shallow(<WelcomeSCR />);
    expect(component).toMatchSnapshot("WelcomeSCR snapshot");
  });
  it('Initial error value should be false in state', () => {
    const wrapper = shallow(<WelcomeSCR />);
    expect(wrapper.instance().state.error).toBeFalsy();
  });
  it('getNews have to be called', async () => {
    const getNewsSpy = jest.fn();
    const wrapper = shallow(<WelcomeSCR />);
    const instance = wrapper.instance();
    instance.getNews();
    await expect(getNewsSpy).toBeDefined();
  });
  it('fetches data from server when server returns a successful response', async () => { // 1
    const mockSuccessResponse = {};
    const mockJsonPromise = Promise.resolve(mockSuccessResponse); // 2
    const mockFetchPromise = Promise.resolve({ // 3
      json: () => mockJsonPromise,
    });
    jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise); // 4
    
    const wrapper = shallow(<WelcomeSCR />); // 5
                            
    await expect(global.fetch).toHaveBeenCalled();
    await expect(global.fetch).toHaveBeenCalledWith(API_REQUEST);
    await expect(wrapper.state().error).toBeFalsy();

    process.nextTick(() => { // 6
      expect(wrapper.state().error).toBeTruthy();
      global.fetch.mockClear(); // 7
    });
    //   done(); // 8
  });
});
