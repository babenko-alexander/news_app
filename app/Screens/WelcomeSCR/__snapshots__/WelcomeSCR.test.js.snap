// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeSCR component should match to snapshot: WelcomeSCR snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WelcomeSCR
    fetchNewsRequest={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text>
          Welcome to the most honest news application!
        </Text>,
        <TouchableOpacity
          activeOpacity={0.7}
          onPress={[Function]}
          style={
            Object {
              "borderRadius": 50,
            }
          }
        >
          <View>
            <Text>
              Get News
            </Text>
          </View>
        </TouchableOpacity>,
      ],
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Welcome to the most honest news application!",
          "style": undefined,
        },
        "ref": null,
        "rendered": "Welcome to the most honest news application!",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.7,
          "children": <View>
            <Text>
              Get News
            </Text>
          </View>,
          "onPress": [Function],
          "style": Object {
            "borderRadius": 50,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text>
              Get News
            </Text>,
            "style": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Get News",
              "style": undefined,
            },
            "ref": null,
            "rendered": "Get News",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text>
            Welcome to the most honest news application!
          </Text>,
          <TouchableOpacity
            activeOpacity={0.7}
            onPress={[Function]}
            style={
              Object {
                "borderRadius": 50,
              }
            }
          >
            <View>
              <Text>
                Get News
              </Text>
            </View>
          </TouchableOpacity>,
        ],
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Welcome to the most honest news application!",
            "style": undefined,
          },
          "ref": null,
          "rendered": "Welcome to the most honest news application!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.7,
            "children": <View>
              <Text>
                Get News
              </Text>
            </View>,
            "onPress": [Function],
            "style": Object {
              "borderRadius": 50,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text>
                Get News
              </Text>,
              "style": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Get News",
                "style": undefined,
              },
              "ref": null,
              "rendered": "Get News",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
